x-pack/plugins/graph/public/state_management/url_templates.ts

```diff
@@ -6,10 +6,10 @@
 
 import actionCreatorFactory from 'typescript-fsa';
 import { reducerWithInitialState } from 'typescript-fsa-reducers/dist';
-import { KibanaParsedUrl } from 'ui/url/kibana_parsed_url';
 import { i18n } from '@kbn/i18n';
 import rison from 'rison-node';
 import { takeEvery, select } from 'redux-saga/effects';
+import { KibanaParsedUrl } from './kibana_parsed_url';
```

Comment:
As this is used in just a single place here and is just taking a single code path through the whole thing, I think it's fine to condense the actual logic and inline it into `generateDefaultTemplate` directly using only the core services and utilities.

Something like (just for illustration):
```ts
const appPath = modifyUrl(`/discover`, parsed => {
      parsed.query['_g'] = rison.encode(...);
    });
const discoverUrl = core.http.basePath.prepend(`app/kibana#${appPath}`)
```

It's already making tons of assumptions about discover URLs anyway. I'll create a separate issue to switch to https://github.com/elastic/kibana/pull/57496 at some point, but for now let's make it as little code as possible in here

Reply:
Thank you for suggestion. I updated the code. 
I also left the call of `parse` function from `url` module. Since `rison.encode` [wraps](https://github.com/w33ble/rison-node/blob/5ca91298ca23e1156ce4f1f233b3d93e75fe1e11/js/rison.js#L206-L213) `index` value into quotes when `id` starts with a number (in our case e.g. `datasource.id` is `90943e30-9a47-11e8-b64d-95841ca0b247`), to replace `'` to `%27` in the url I partially left the implementation of [`prependPath`](https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/url/prepend_path.ts#L31).